charts/gardener/controlplane/charts/runtime/templates/apiserver/hvpa.yaml

```diff
@@ -0,0 +1,73 @@
+{{- if .Values.global.apiserver.hvpa.enabled }}
+{{- /* .Values.global.apiserver.replicaCount is of type 'float64', so let's cast it to string to have proper types for comparison */}}
+{{- if ne (.Values.global.apiserver.replicaCount | toString) "0" }}
+apiVersion: autoscaling.k8s.io/v1alpha1
+kind: Hvpa
+metadata:
+  name: gardner-apiserver-hvpa
+  namespace: garden
+spec:
+  replicas: 1
+{{- if .Values.global.apiserver.hvpa.scaleUpStabilization }}
+  scaleUpStabilization:
+{{ toYaml .Values.global.apiserver.hvpa.scaleUpStabilization | indent 4 }}
+{{- end }}
+{{- if .Values.global.apiserver.hvpa.scaleDownStabilization }}
+  scaleDownStabilization:
+{{ toYaml .Values.global.apiserver.hvpa.scaleDownStabilization | indent 4 }}
+{{- end }}
+  hpa:
+    selector:
+      matchLabels:
+        role: gardener-apiserver-hpa
+    template:
+      metadata:
+        labels:
+          role: gardener-apiserver-hpa
+      spec:
+        maxReplicas: {{ .Values.global.apiserver.hvpa.maxReplicas }}
+        minReplicas: {{ .Values.global.apiserver.hvpa.minReplicas }}
+        metrics:
+        - resource:
+            name: memory
+            targetAverageUtilization: {{ .Values.global.apiserver.hvpa.targetAverageUtilization }}
+          type: Resource
+        - resource:
+            name: cpu
+            targetAverageUtilization: {{ .Values.global.apiserver.hvpa.targetAverageUtilization }}
+          type: Resource
+    updatePolicy:
+      updateMode: "Auto"
+  vpa:
+    selector:
+      matchLabels:
+        role: gardener-apiserver-vpa
+    template:
+      metadata:
+        labels:
+          role: gardener-apiserver-vpa
+      spec:
+        resourcePolicy:
+          containerPolicies:
+            - containerName: gardener-apiserver
+              maxAllowed:
+                memory: 25G
+                cpu: "8"
+              minAllowed:
+                memory: 400M
+                cpu: 400m
+    updatePolicy:
+      updateMode: "Auto"
+  weightBasedScalingIntervals:
+    - vpaWeight: 0
+      startReplicaCount: {{ .Values.global.apiserver.hvpa.minReplicas }}
+      lastReplicaCount: {{ .Values.global.apiserver.hvpa.lastReplicaCountForHpa }}
+    - vpaWeight: 100
+      startReplicaCount: {{ .Values.global.apiserver.hvpa.maxReplicas }}
```

Comment:
Shouldn't this be something like: `{{ .Values.global.apiserver.hvpa.lastReplicaCountForHpa + 1 }}`?

Reply:
currently `lastReplicaCountForHpa` = `maxReplicas - 1`
I am not sure if there is a math operator so that this can be done in the chart itself.

Reply:
See https://github.com/gardener/gardener/blob/master/charts/gardener/controlplane/charts/runtime/templates/apiserver/poddisruptionbudget.yaml#L14

Reply:
Thanks Rafael, this helps!

Reply:
Though this might be an issue if `maxReplicas` is 1 or equal to `minReplicas`. What do you suggest?

Reply:
I guess, in that case we would want to scale only vertically, right?
So we could just wrap the first interval in an if-block...

Reply:
right. Done changes. Thanks Tim.